rust/lance/src/io/exec/take.rs

```diff
@@ -76,6 +76,7 @@ impl Take {
                     };
                     Ok::<RecordBatch, Error>(rows)
                 })
+                .buffered(num_cpus::get())
```

Comment:
This didn't seem to have much benefit locally, but I suspect it might when the object store has higher latency.